### ISSUE TITLE
Bufferline highlighting

### ISSUE THREAD

**Issue by akinsho** [NTBBloodbath/doom-one.nvim#8]
@NTBBloodbath I just noticed the changes to the bufferline highlighting, I just thought it was worth noting that I actually designed bufferline to be agnostic of colorschemes where possible.

I'm not sure where or how it started but colorscheme authors have started overriding the plugins internal highlights which I never exposed on purpose since it is not really what I intended. I would have made them private if possible, so the only way to change them was a user using their plugin config. The idea with that plugin is that someone could change between like 10 colorschemes and their bufferline would look roughly the same, but now all sorts of different colours are applied by different themes.

Can this change please be optional, since I'd rather at least the option of things behaving normally. 

For example, my bufferline now looks like

![image](https://user-images.githubusercontent.com/22454918/126280533-e4ac5917-b447-4b4c-b261-62f37a67fa5e.png)

The separator colour is greenish and the bufferline fill is changed to be the same as the tabs themselves which breaks the effect of it being like a darker shadowed background (which was the original intent)

Lol sorry for the long spiel, definitely something I've been thinking about a lot with that plugin and just happened to land here since this is the theme I use.

**Comment by NTBBloodbath** [comment]
Hey, this was intended to fix the issue #7 so I can maybe link the colors override with the transparent background option :thinking:

Or do you have any other idea to fix that issue? To be honest, I don't like overriding plugin colors so much (that's why there are so few supported and not a long list), however I didn't find another solution for that problem and I don't know much about the bufferline codebase.

Regards

**Comment by akinsho** [comment]
@NTBBloodbath I'll have read through that issue, I think the problem is that bufferline expects a user to have a `Normal` highlight but people who use transparent backgrounds clear that group and bufferline doesn't handle that. Tbh I'm not immediately sure how to handle that in the plugin but will have a think. It might be that it can't be supported without them setting some highlights themselves in that case.

TLDR: transparent backgrounds aren't really supported by bufferline currently, probably without a user tweaking them.

**Comment by NTBBloodbath** [comment]
I have changed this behavior to override bufferline coloring **only** when using a transparent background as a temporal workaround for it. Let me know in the future if bufferline has support for transparent backgrounds to remove that block of code :)

Regards!

**Comment by akinsho** [comment]
Thanks, @NTBBloodbath, I'll look into the transparent behaviour. Tbh I didn't really think many people with a transparent background would use bufferline since IMO it sort of defeats the point 😆. This is a good interim solution till I figure out an official solution in bufferline.
